presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueriesWithExchangeMaterialization.java

```diff
@@ -171,6 +171,18 @@ private void testBucketedByHiveUnsupportedTypeForTemporaryTable(
         assertUpdate("DROP TABLE IF EXISTS test_materialize_bucket_by_non_hive_types");
     }
 
+    // We need to disable optimize_constant_grouping_keys so Optimizer does NOT optimize
+    // [Aggregate() -> Project(add constant) -> TableScan()] ==> [Project(add constant) -> Aggregate() -> TableScan()]
+    // which will lead to type changes in materialized tables
+    private Session.SessionBuilder sessionBuilderNoConstantGrouping()
```

Comment:
Which changes exactly causing these behavior change?

Reply:
See the reply to Rebecca above

Reply:
IntelliJ git review does not show previous comments, thats why I missed. :-(